numpy/core/tests/test_api.py

```diff
@@ -38,58 +38,39 @@ def test_array_array():
         np.array(np.array(1.0, dtype=np.float64), dtype=np.float64)
         assert_equal(old_refcount, sys.getrefcount(np.float64))
 
+    # test string
+    B2 = np.dtype((bytes, 2))
+    B3 = np.dtype((bytes, 3))
+    B5 = np.dtype((bytes, 5))
+    assert_equal(np.array(b"1.0", dtype=np.float64),
+                 np.ones((), dtype=np.float64))
+    assert_equal(np.array(b"1.0").dtype, B3)
+    assert_equal(np.array(b"1.0", dtype=bytes).dtype, B3)
+    assert_equal(np.array(b"1.0", dtype=B2), np.array(b"1."))
+    assert_equal(np.array(b"1", dtype=B5), np.ones((), dtype=B5))
+
     # test string
     S2 = np.dtype((str, 2))
     S3 = np.dtype((str, 3))
     S5 = np.dtype((str, 5))
```

Comment:
```suggestion
    U5 = np.dtype((str, 5))
```
Same reasoning as above